buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -565,10 +565,25 @@ static ByteBuf encodeString0(ByteBufAllocator alloc, boolean enforceHeap, CharBu
         }
     }
 
+    static String readUtf8(AbstractByteBuf buf, int readerIndex, int length) { // Fast-Path for a AbstractByteBuf
+        buf.checkIndex(readerIndex, length);
+        UTF8Decoder.UTF8Processor processor = new UTF8Decoder.UTF8Processor(new char[length]);
```

Comment:
If you don't want to bother caching the processor itself, that's the most straightforward manner, yes.